### PR TITLE
feat(trace): Add chainID and version fields in trace.

### LINKED ISSUE #219: feature requests on trace data structure
listed below:

### PR DESCRIPTION
**Pull request by mask-pp**
Add `blockTrace.chainID` and `blockTrace.version` fields in trace struct.
The version changed depend on the current tag， such as `alpha-v1.3` => `1.5.0-alpha`.

This PR should wait for [Update go version](https://github.com/scroll-tech/go-ethereum/pull/241) before merge.

### CODE REVIEW

**Review by 0xmountaintop — DISMISSED** [review]

**Comment by lispc** [comment]
what will version be for untagged commit?  

like "1.8.13-unstable-21c059b6"?

==

after reading code, i guess the version will always be 1.8.13/1.9.20/1.10.5 **instead of** 1.8.13-alpha..  
i am afraid this version string is not clear enough.  Say we have a trace with version 1.3.0, we will not know the exact sequencer code commit to generate this trace.

i think use `VersionWithCommit` will be a better choice. @HAOYUatHZ @mask-pp

**Comment by mask-pp** [comment]
> 

Changed to `1.3.0-alpha-e10a9e33` type.

**Comment by HAOYUatHZ** [comment]
if you do it this way you need to update Makefile

**Review by lispc — DISMISSED** [review]

File: params/version.go [hunk 1]
```diff
@@ -18,13 +18,14 @@ package params
...
 )
 
 const (
-	VersionMajor = 1        // Major version component of the current release
-	VersionMinor = 10       // Minor version component of the current release
-	VersionPatch = 13       // Patch version component of the current release
-	VersionMeta  = "stable" // Version metadata to append to the version string
+	VersionMajor = 1       // Major version component of the current release
+	VersionMinor = 3       // Minor version component of the current release
+	VersionPatch = 0       // Patch version component of the current release
```
Comment (0xmountaintop) [inline params/version.go]:
```suggestion
	VersionMajor = 0       // Major version component of the current release
	VersionMinor = 1       // Minor version component of the current release
	VersionPatch = 5       // Patch version component of the current release
```

**Comment by lispc** [comment]
btw after this PR merged, i think it is time to regenerate traces in scroll-zkevm? @HAOYUatHZ        (or directly make a new "test-traces" repo? and use it as git sub module inside scroll-zkevm? ) not of high priority, but needed before merge goerli-0215 into main)

**Comment by 0xmountaintop** [comment]
> btw after this PR merged, i think it is time to regenerate traces in scroll-zkevm? @HAOYUatHZ (or directly make a new "test-traces" repo? and use it as git sub module inside scroll-zkevm? ) not of high priority, but needed before merge goerli-0215 into main)

ok

File: eth/tracers/api_blocktrace.go [hunk 2]
```diff
@@ -357,6 +358,8 @@ func (api *API) fillBlockTrace(env *traceEnv, block *types.Block) (*types.BlockT
 	}
 
 	blockTrace := &types.BlockTrace{
+		ChainID: api.backend.ChainConfig().ChainID.Uint64(),
+		Version: fmt.Sprintf("%s-%s", params.VersionWithMeta, params.CommitHash[:8]),
```
Comment (0xmountaintop) [inline eth/tracers/api_blocktrace.go]:
```suggestion
		Version: fmt.Sprintf("%s-%s", params.VersionWithMeta(), params.CommitHash()[:8]),
```

Comment (0xmountaintop) [inline eth/tracers/api_blocktrace.go]: and I prefer 

```suggestion
		Version: params.ArchiveVersion(params.CommitHash())
```

Comment (mask-pp) [reply eth/tracers/api_blocktrace.go]: `params.VersionWithMeta` and `params.CommitHash` are `func() string` type, so don't need to change.

File: eth/tracers/api_blocktrace.go [hunk 3]
```diff
@@ -357,6 +358,8 @@ func (api *API) fillBlockTrace(env *traceEnv, block *types.Block) (*types.BlockT
 	}
 
 	blockTrace := &types.BlockTrace{
+		ChainID: api.backend.ChainConfig().ChainID.Uint64(),
+		Version: params.ArchiveVersion(params.CommitHash),
```
Comment (0xmountaintop) [inline eth/tracers/api_blocktrace.go:362]:
```suggestion
		Version: params.ArchiveVersion(params.CommitHash()),
```

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**